api/src/wfl/executor.clj

```diff
@@ -114,19 +113,27 @@
     (when-not (= "importSnapshot" fromSource)
       (throw
        (UserException. "Unsupported coercion" (util/make-map fromSource))))
-    (throw-on-method-config-version-mismatch
-     (method-config-or-throw workspace methodConfiguration)
-     methodConfigurationVersion))
+    (let [m (firecloud/method-configuration workspace methodConfiguration)]
+      (throw-on-method-config-version-mismatch m methodConfigurationVersion)
+      (throw-unless-dockstore-method (:methodRepoMethod m))))
   executor)
 
+(defn ^:private entity-from-snapshot
+  "Coerce the `snapshot` into a workspace entity via `fromSource`."
+  [{:keys [workspace fromSource] :as executor} {:keys [id name] :as snapshot}]
+  (when-not (= "importSnapshot" fromSource)
+    (throw (ex-info "Unknown conversion from snapshot to workspace entity."
+                    {:executor executor
+                     :snapshot snapshot})))
+  (rawls/create-snapshot-reference workspace id name))
+
 (defn ^:private from-source
-  "Coerce `source-item` to form understood by `executor` via `fromSource`."
-  [{:keys [workspace fromSource] :as executor}
-   {:keys [name id]              :as _source-item}]
-  (cond (= "importSnapshot" fromSource)
-        (rawls/create-or-get-snapshot-reference workspace id name)
-        :else
-        (throw (ex-info "Unknown fromSource" {:executor executor}))))
+  "Coerce `object` to form understood by `executor``."
+  [executor [description item :as object]]
+  (cond (= "snapshot" description) (entity-from-snapshot executor item)
+        :else                      (throw (ex-info "No method to coerce object into workspace entity"
+                                                   {:executor executor
+                                                    :object   object}))))
```

Comment:
I'll take responsibility here, I added this method initially.  I liked the `cond` because I was pretty sure we'd be supporting other methods of coercion in the future.

Reply:
I'm trying to program intent. So far we only support snapshots. If we did https://docs.google.com/presentation/d/159ZnFwWh2tCTv4e9n3IxnGkC5VNfMorO4FqKJpjXuyY/edit#slide=id.ge1e3db2ca4_0_0, we'd have a terra workspace source that would supply `entity`s and so we wouldn't require any "coercion".

Reply:
Used a case analysis.